beaker_kernel/lib/context.py

```diff
@@ -32,23 +37,25 @@ class BaseContext:
 
     WEIGHT: int = 50  # Used for auto-sorting in drop-downs, etc. Lower weights are listed earlier.
 
-    def __init__(self, beaker_kernel: "LLMKernel", subkernel: "BaseSubkernel", agent_cls: "BaseAgent", config: Dict[str, Any]) -> None:
+    def __init__(self, beaker_kernel: "LLMKernel", agent_cls: "BaseAgent", config: Dict[str, Any]) -> None:
         self.intercepts = []
         self.jinja_env = None
         self.templates = {}
         self.beaker_kernel = beaker_kernel
-        self.subkernel = subkernel
         self.agent = agent_cls(
             context=self,
             tools=[],
         )
         self.config = config
+        self._init_subkernel()
```

Comment:
Generally, it's best to define all of the instance attributes in the main __init__ method. If you start defining them in other methods/funcs/files, it can get very hard to figure out where in the code actual initialization happens, and this is still initialization.

I'd recommend something like:
```suggestion
        self.subkernel = self.get_subkernel()
```